app/src/main/java/chat/rocket/android/authentication/login/presentation/LoginPresenter.kt

```diff
@@ -1,51 +1,59 @@
 package chat.rocket.android.authentication.login.presentation
 
-import chat.rocket.android.authentication.infraestructure.AuthTokenRepository
+import android.widget.EditText
 import chat.rocket.android.authentication.presentation.AuthenticationNavigator
 import chat.rocket.android.core.lifecycle.CancelStrategy
 import chat.rocket.android.util.launchUI
+import chat.rocket.android.util.textContent
 import chat.rocket.common.RocketChatException
 import chat.rocket.common.RocketChatTwoFactorException
-import chat.rocket.common.util.PlatformLogger
 import chat.rocket.core.RocketChatClient
+import chat.rocket.core.internal.rest.chatRooms
+import chat.rocket.core.internal.rest.getRoomFavoriteMessages
 import chat.rocket.core.internal.rest.login
-import okhttp3.HttpUrl
-import okhttp3.OkHttpClient
+import chat.rocket.core.internal.rest.sendMessage
+import timber.log.Timber
 import javax.inject.Inject
 
 class LoginPresenter @Inject constructor(private val view: LoginView,
                                          private val strategy: CancelStrategy,
-                                         private val navigator: AuthenticationNavigator,
-                                         private val okHttpClient: OkHttpClient,
-                                         private val logger: PlatformLogger,
-                                         private val repository: AuthTokenRepository) {
+                                         private val navigator: AuthenticationNavigator) {
+    @Inject lateinit var client: RocketChatClient
 
-    val client: RocketChatClient = RocketChatClient.create {
-        httpClient = okHttpClient
-        restUrl = HttpUrl.parse(navigator.currentServer)!!
-        websocketUrl = navigator.currentServer!!
-        tokenRepository = repository
-        platformLogger = logger
-    }
-
-    fun authenticate(username: String, password: String) {
-        // TODO - validate input
-
-        launchUI(strategy) {
-            view.showLoading()
-            try {
-                val token = client.login(username, password)
+    fun authenticate(usernameOrEmailEditText: EditText, passwordEditText: EditText) {
```

Comment:
The presenter should never have access to Android views (widgets), ideally the presenter will never touch any Android specific code.

Reply:
I agree that the presenter should never have access/dependencies to the Android API's. The presenter should be easily testable. I'll change it.